lib/excoveralls/cobertura.ex

```diff
@@ -159,12 +159,13 @@ defmodule ExCoveralls.Cobertura do
   end
 
   defp module_name(source) do
-    case Regex.run(~r/^defmodule\s+(.*)\s+do$/m, source, capture: :all_but_first) do
+    case Regex.run(~r/^def(?:module|protocol|impl)\s+(.*)\s+do$/m, source, capture: :all_but_first) do
+
       [module] ->
         module
 
       _ ->
-        [module] = Regex.run(~r/^-module\((.*)\)\.$/m, source, capture: :all_but_first)
+        [module] = Regex.run(~r/^-(?:module|protocol|impl)\((.*)\)\.$/m, source, capture: :all_but_first)
         module
     end
```

Comment:
A couple of comments here:
1. erlang does not have `-protocol(...)` and `-impl(...)` so you don't need to add them on the second regex.
2. due to Elixir extensibility we can't be sure a file contains at least one of `defmodule` , `defprotocol`, `defimpl` so you should also add a fallback clause in case extracting the module name is not possible.

Something like this should work:


```elixir
with nil <- Regex.run(~r/^def(module|protocol|impl)\s+(.*)\s+do$/m, source, capture: :all_but_first),
     nil <- Regex.run(~r/^-module\((.*)\)\.$/m, source, capture: :all_but_first) do
  "UNKNOWN_MODULE"
else
  [module] -> module
end       
```

Reply:
1. You’ve learnt me some erlang ;)
2. thanks, I’ve updated the PR according to your feedback

Thank you for your review!